lib/elixir/lib/code/formatter.ex

```diff
@@ -521,14 +521,15 @@ defmodule Code.Formatter do
               {string(~S{"..//":}), state}
 
             {:__block__, _, [atom]} when is_atom(atom) ->
-              key =
+              iodata =
                 if Macro.classify_atom(atom) in [:identifier, :unquoted] do
-                  IO.iodata_to_binary([Atom.to_string(atom), ?:])
+                  [Atom.to_string(atom), ?:]
                 else
-                  IO.iodata_to_binary([?", Atom.to_string(atom), ?", ?:])
+                  [?", atom |> Atom.to_string() |> String.replace("\"", "\\\""), ?", ?:]
```

Comment:
I think we should use `Macro.inspect_atom(:key, atom)`. This way we can even skip the `Macro.classify_atom` call.

Reply:
Oh that's a nice one, totally forgot about inspect_atom!

But it seems to be over-escaping backslashes for our usage:
<img width="671" alt="Screenshot 2023-11-15 at 20 55 05" src="https://github.com/elixir-lang/elixir/assets/11598866/b44f3cfc-3c9d-4152-87e5-7c85bc2c421b">

Will try to figure out if we can remove the double escape.

Reply:
Also noticed that this code [here](https://github.com/elixir-lang/elixir/blob/main/lib/elixir/lib/code/formatter.ex#L1583) is almost the same, there's probably a way to refactor it (maybe with `Macro.inspect_atom/2`)

Reply:
Ah, it is because strings in the formatted already unescaped. My bad. Your PR is good to go!